docs/Inputs.md

```diff
@@ -509,7 +509,7 @@ import { RadioButtonGroupInput, ReferenceInput } from 'react-admin'
 
 ## `<ReferenceInput>`
 
-Use `<ReferenceInput>` for foreign-key values, i.e. to let users choose a value from another REST endpoint. This component fetches the possible values in the reference resource (using the `GET_LIST` REST method), then delegates rendering to a subcomponent, to which it passes the possible choices as the `choices` attribute.
+Use `<ReferenceInput>` for foreign-key values, i.e. to let users choose a value from another REST endpoint. This component fetches the possible values in the reference resource (using the `GET_LIST` REST method) and the referenced record (using the `GET_ONE` REST method), then delegates rendering to a subcomponent, to which it passes the possible choices as the `choices` attribute and the referenced record as `selectedItem`.
```

Comment:
The value is available and it saves computation, so I don't understand why it isn't passed on. I think relying on a `selectedItem` being passed on vs looking up the value yourself using the choices array just makes more sense imo.

Reply:
You didn't understand me. This value is already passed in the form of the `choices` and `input.value`, so you have to give me a proper use case for passing it twice. It's not needed in `SelectInput` or `AutoCompleteInput`, and I don't understand why it would be needed in any other similar component. 

I'm ready to change my opinion about this if you provide a strong use cas for it. But as it delays the merge of this PR, I strongly suggest that you remove all `selectedItem` related code from this PR.

Reply:
Yes I understand, here goes my strong usecase 😄;  

Consider `ReferenceInput` wrapping `AutocompleteInput` and a resource is referenced with 100.000 records, you don't want to retrieve the entire resultset, so you'll return a limited subset (the first 25 or so) so the user has something to select, but if this subset does not include the record of `input.value` the selectedItem can't be resolved from the choices! It's possible to always include the selectedItem in the choices, by changing the selector; but then the user will always see the current value in the selection items and this could lead to unexpected results. For example the current input value is resolved to "France" and the user types "Nether" and the dropdown still shows "France" as viable option. 

This is solved by passing through `selectedItem`. Btw; this only makes sense for `Autocomplete`, because for SelectInput all records of the resource should be resolved.

Reply:
This case is already handled by the `getPossibleReferences` selector in the `references/possibleValues` reducer. So the selected item is always in the choices, whatever the number of actual choices. 

Do you have another valid use case?

Reply:
Last attempt; it does handle it, but I am arguing it has an unwanted side effect. It will always show the current value in the suggestion list. I think this is counter intuitive. 
![2018-02-02_16-30-34](https://user-images.githubusercontent.com/621098/35740917-962e0108-0836-11e8-8a92-2a5284ea5a0d.gif)
This example shows I am looking for "Illum veri", but I get the current value in the suggestion list. Filtering the choices list (to remove input.value) before showing the suggestions isn't an option either, because if I type "Perspi" it should be included in the choices list. 

If this doesn't convince you, I'll remove it from the PR.

Reply:
I agree that it may be counterintuitive, but both the SelectInput and the AutocompleteInput need the current value to be in the choices in their current implementation. So to remove the current choice in the GUI, you need to change not only the ReferenceItem component, but also its potential child components. Changing these components is very not BC, and IMO way beyond the purpose of this PR. 

So no, I’m not convinced.

Reply:
Ok, let's put a pin in it. No trace of `selectedItem` in the PR anymore. https://github.com/marmelab/admin-on-rest/pull/1420/commits/14fb0e8568da94b83925383b74645204757bb638

Reply:
There are still traces of it, and there are still comments that you haven't addressed. Please review your own PR using the "Files changed" tab, since the discussion tab is too long to be useful (another good reason to make small and atomic PRs by the way).